modules/samtools/bamtocram/main.nf

```diff
@@ -0,0 +1,35 @@
+//There is a -L option to only output alignments in interval, might be an option for exons/panel data?
+process SAMTOOLS_BAMTOCRAM {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::samtools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.15.1--h1170115_0' :
+        'quay.io/biocontainers/samtools:1.15.1--h1170115_0' }"
+
+    input:
+    tuple val(meta), path(input), path(index)
+    path  fasta
+    path  fai
+
+    output:
+    tuple val(meta), path("*.cram"), path("*.crai"), emit: cram_crai
+    path  "versions.yml"                           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args  ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    samtools view --threads ${task.cpus-1} --reference ${fasta} -C $args $input > ${prefix}.cram
+    samtools index -@${task.cpus} ${prefix}.cram
```

Comment:
From a sarek perspective I like having this here, but from a modules one, I am not so sure if it should be in one module. On the other hand we also have `collate|bam2fastq` in one module 🤔

Reply:
:eyes: I thought maybe in this case of indexing in the same module (After the big bust up a year ago ;) )? But I would also personally keep it separate

Reply:
I think that if we use it in the sarek preprocessing, it might be used at some point by other pipelines, so I'd go for a nf-core module over a local one in this case